plugins/cache/cache_main.ml

```diff
@@ -247,12 +256,15 @@ let create reader writer =
             "entry" ".cache" in
         Data.Write.to_channel writer ch proj;
         Out_channel.close ch;
-        {
-          index with
-          entries = Map.set index.entries ~key:id ~data:{
-              size = size path; path; atime = ctime; ctime; hits = 1
-            }
-        }) in
+        let entry = {
+          size = size path; path; atime = ctime; ctime; hits = 1
+        } in
```

Comment:
I don't like performing a removal as a side-effect of  a pure function. It is better to express it explicitly, e.g.,
```ocaml
if Map.mem index.entries id then begin
   warning "overwriting an entry";
   Index.remove_entry e;
end
```
Also, in general, this PR doesn't follow the spirit of the implementation, where the entries are removed all in one place. This is assumed in the code so I'm not sure whether this change breaks this assumptions anywhere.